### PR TITLE
wire printflags through set commands

### PR DESCRIPTION
**Pull request by juanvallejo**
**Release note**:
```release-note
NONE
```

~~Depends on https://github.com/kubernetes/kubernetes/pull/62139
Opening now to test with CI~~

Wires PrintFlags through all `kubectl set` commands.

cc @soltysh @deads2k

### CODE REVIEW

File: pkg/kubectl/cmd/set/set_env.go [hunk 1]
```diff
@@ -215,7 +215,13 @@ func (o *EnvOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []stri
 	o.Resources = resources
 	o.Cmd = cmd
 
-	o.PrintFlags.Complete(o.DryRun)
+	if o.DryRun {
+		// TODO(juanvallejo): This can be cleaned up even further by creating
+		// a PrintFlags struct that binds the --dry-run flag, and whose
+		// ToPrinter method returns a printer that understands how to print
+		// this success message.
+		o.PrintFlags.Complete("%s (dry run)")
```
Comment (juanvallejo) [inline pkg/kubectl/cmd/set/set_env.go]: @deads2k updated the Complete method to receive a string template adding the "dry run" bits to the message.

File: pkg/kubectl/cmd/set/set_image.go [hunk 2]
```diff
@@ -232,18 +250,18 @@ func (o *ImageOptions) Run() error {
 			continue
 		}
 
-		// no changes
```
Comment (soltysh) [inline pkg/kubectl/cmd/set/set_image.go]: Don't change the order here. Open a separate pr addressing this for all set commands, like it was already mentioned in #62139

File: pkg/kubectl/cmd/set/set_selector.go [hunk 3]
```diff
@@ -52,6 +56,8 @@ type SelectorOptions struct {
 	PrintObject      func(obj runtime.Object) error
```
Comment (soltysh) [inline pkg/kubectl/cmd/set/set_selector.go]: That should be removed given you're adding `PrintObj`.

File: pkg/kubectl/cmd/set/set_selector.go [hunk 4]
```diff
@@ -109,6 +119,8 @@ func (o *SelectorOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args [
 	o.dryrun = cmdutil.GetDryRunFlag(cmd)
 	o.output = cmdutil.GetFlagString(cmd, "output")
 
+	o.PrintFlags.Complete(o.dryrun)
```
Comment (soltysh) [inline pkg/kubectl/cmd/set/set_selector.go]: Keep `Complete` and `ToPrinter` next to one another like in the previous set commands.

File: pkg/kubectl/cmd/set/set_selector.go (hunk 3 above)
Comment (soltysh) [reply pkg/kubectl/cmd/set/set_selector.go]: And use the latter for consistency.

File: pkg/kubectl/cmd/set/set_subject.go [hunk 5]
```diff
@@ -70,11 +74,13 @@ type SubjectOptions struct {
 	Groups          []string
 	ServiceAccounts []string
 
-	PrintObject func(obj runtime.Object, out io.Writer) error
+	PrintObject func(obj runtime.Object) error
```
Comment (soltysh) [inline pkg/kubectl/cmd/set/set_subject.go]: `PrintObj` for consistency

**Review by soltysh — CHANGES_REQUESTED** [review]
Overall this lgtm, but there are some nits that needs addressing before merging.

File: pkg/kubectl/cmd/set/set_env.go [hunk 6]
```diff
@@ -215,7 +215,13 @@ func (o *EnvOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []stri
 	o.Resources = resources
 	o.Cmd = cmd
 
-	o.PrintFlags.Complete(o.DryRun)
+	if o.DryRun {
+		// TODO(juanvallejo): This can be cleaned up even further by creating
```
Comment (soltysh) [inline pkg/kubectl/cmd/set/set_env.go]: Hmmm... this is worth a bigger discussion. @deads2k I'm slightly skeptical adding a whole Printer just to pass the `(dry run)` message, seems a slight overkill. Why not just embed `DryRun` inside the printer, but leave the wiring to a user? Let's discuss this on Monday.

File: pkg/kubectl/cmd/set/set_image.go (hunk 2 above)
Comment (juanvallejo) [reply pkg/kubectl/cmd/set/set_image.go]: Ah, completely missed reverting this change on this file. Will sweep through all other files and see if I missed reverting this elsewhere. Thanks

**Review by soltysh — APPROVED** [review]
/lgtm
/approve no-issue

**Pull request merged**